processor/runtime/src/main/java/com/redhat/service/smartevents/processor/actions/input/InputActionInvoker.java

```diff
@@ -0,0 +1,85 @@
+package com.redhat.service.smartevents.processor.actions.input;
+
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.redhat.service.smartevents.infra.auth.AbstractOidcClient;
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.CloudEventDeserializationException;
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.CloudEventSerializationException;
+import com.redhat.service.smartevents.processor.actions.ActionInvoker;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.ext.web.client.HttpRequest;
+import io.vertx.mutiny.ext.web.client.WebClient;
+
+public class InputActionInvoker implements ActionInvoker {
+
+    private static final Logger LOG = LoggerFactory.getLogger(InputActionInvoker.class);
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private final String endpoint;
+    private final String cloudEventType;
+    private final WebClient client;
+    private final AbstractOidcClient oidcClient;
+
+    public InputActionInvoker(String endpoint, String cloudEventType, WebClient client, AbstractOidcClient oidcClient) {
+        this.endpoint = endpoint;
+        this.cloudEventType = cloudEventType;
+        this.client = client;
+        this.oidcClient = oidcClient;
+    }
+
+    @Override
+    public boolean requiresCloudEvent() {
+        return false;
+    }
+
+    @Override
+    public void onEvent(String event) {
+        LOG.info("InputActionInvoker::onEvent | {} - {} | {}", endpoint, cloudEventType, event);
+
+        HttpRequest<Buffer> request = client.postAbs(endpoint);
+        String token = oidcClient.getToken();
+        LOG.info("InputActionInvoker::onEvent | Token: {}", token);
+        if (token != null && !"".equals(token)) {
```

Comment:
The token is always required as we're sending to the ingress which requires auth.